### PR TITLE
Add support for V8 intrinsics via Babel

### PR DESCRIPTION
**Pull request by RReverser**
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/postprocess.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ const { getLast } = require("../common/util");
 function postprocess(ast, options) {
   visitNode(ast, node => {
     switch (node.type) {
+      case "V8IntrinsicIdentifier":
+        node.name = `%${node.name}`;
+        node.type = "Identifier";
```
Comment (lydell) [inline src/language-js/postprocess.js]: Hmm, are we sure that we want to "lie" to all code that this is a regular identifier? Will this bite us in the future?

Comment (RReverser) [reply src/language-js/postprocess.js]: I'm not 100% sure, but it's meant to work as an identifier for pretty much all purposes... Unless something relies on which chars it contains.

Comment (RReverser) [reply src/language-js/postprocess.js]: So what's the conclusion? Can we proceed as-is? Note that even if there are some issues, they would appear only on code using such intrinsics, so in any case this is not regressing regular JavaScript.

Comment (vjeux) [reply src/language-js/postprocess.js]: I chatted offline with @RReverser and he is going to send another version that doesn’t turn the intrinsic into an identifier during post processing but instead update the regular printer to handle intrinsics properly.

Once this is done, we should be good to accept.

On a side note, it’s really cool to see people at Google starting to play with prettier and contribute back. AMP and now v8 :)

File: yarn.lock [hunk 2]
```diff
@@ -282,9 +282,10 @@
     esutils "^2.0.2"
     js-tokens "^4.0.0"
 
-"@babel/parser@7.2.0":
-  version "7.2.0"
-  resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.2.0.tgz#02d01dbc330b6cbf36b76ac93c50752c69027065"
+"@babel/parser@7.6.0", "@babel/parser@^7.6.0":
```
Comment (vjeux) [inline yarn.lock]: We should do the upgrade in a different PR and fix the few tests that are breaking as a result.

File: tests/v8_intrinsic/intrinsic_call.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function doSmth()     {
+            %DebugPrint
+        (
+                foo )
+  }
```
Comment (vjeux) [inline tests/v8_intrinsic/intrinsic_call.js]: Could you expand the test coverage to have other places where v8 intrinsics are used? One test seems light.

File: src/language-js/postprocess.js (hunk 1 above)
Comment (RReverser) [reply src/language-js/postprocess.js]: I've updated the PR to preserve node as-is. Unfortunately, there are still failures related to upgrade of `@babel/parser` - I'd be really grateful if that upgrade could be done by someone who is more familiar with the codebase :)

File: src/language-js/printer-estree-json.js [hunk 4]
```diff
@@ -52,6 +52,8 @@ function genericPrint(path, options, print) {
       return JSON.stringify(node.value);
     case "Identifier":
       return JSON.stringify(node.name);
+    case "V8IntrinsicIdentifier":
+      return JSON.stringify(`%${node.name}`);
```
Comment (vjeux) [inline src/language-js/printer-estree-json.js]: Can there be intrinsics in json files?

File: src/language-js/printer-estree.js [hunk 5]
```diff
@@ -766,6 +766,8 @@ function printPathNoParens(path, options, print, args) {
         printTypeAnnotation(path, options, print)
       ]);
     }
+    case "V8IntrinsicIdentifier":
```
Comment (vjeux) [inline src/language-js/printer-estree.js]: Did you grep the file for “Identifier” to see if there are others to update?

File: src/language-js/printer-estree-json.js (hunk 4 above)
Comment (RReverser) [reply src/language-js/printer-estree-json.js]: No. I guess I misunderstood what this file does, I thought it's printing out arbitrary ESTree ASTs as JSON.

File: src/language-js/printer-estree.js (hunk 5 above)
Comment (RReverser) [reply src/language-js/printer-estree.js]: Yes.

Comment (RReverser) [reply src/language-js/printer-estree.js]: (others don't seem relevant, and, as turns out, even the JSON one is not)

File: tests/v8_intrinsic/intrinsic_call.js (hunk 3 above)
Comment (RReverser) [reply tests/v8_intrinsic/intrinsic_call.js]: Added few more, although it's not very easy to be creative with something like this :)

**Comment by sosukesuzuki** [comment]
Updating babylon seems to be on #6386 (but I don't know the status).
Probably, we should add it to 1.19 milestone (or create new PR to update babylon)

**Comment by fisker** [comment]
@RReverser @sosukesuzuki 

Unfortunately, https://github.com/prettier/prettier/pull/6386 is blocked  by https://github.com/babel/babel/pull/10220 and 7.6.0 still  nots olve  that  problem yet

**Comment by RReverser** [comment]
Rebased onto master and fixed conflicts for now.

**Comment by fisker** [comment]
@RReverser babylon is up to date #6386

**Comment by RReverser** [comment]
Yup, in progress.

**Comment by RReverser** [comment]
Done.

**Comment by vjeux** [comment]
Yay!’
